Packs/Base/Scripts/DBotMLFetchData/DBotMLFetchData.py

```diff
@@ -1,6 +1,1031 @@
+    if len(headers_with_name) == 0:
+        return None
+    else:
+        return headers_with_name[index]['value']
+
+
+def get_headers_features(email_headers):
+    spf_result = get_header_value(email_headers, header_name='Received-SPF')
+    if spf_result is not None:
+        spf_result = spf_result.split()[0].lower()
+    else:
+        spf_result = 'other'
+    res_spf = {k: 0 for k in ['none', 'neutral', 'pass', 'fail', 'softfail', 'other', 'temperror', 'permerror']}
+    if spf_result in res_spf:
+        res_spf[spf_result] += 1
+    else:
+        res_spf['other'] += 1
+    res_spf['non-positive'] = spf_result in set(['none', 'fail', 'softfail'])
+    authentication_results = get_header_value(email_headers, header_name='Authentication-Results')
+    if authentication_results is not None:
+        if 'dkim=' in authentication_results:
+            dkim_result = authentication_results.split('dkim=')[-1].split()[0].lower().strip()
+        else:
+            dkim_result = 'other'
+    else:
+        dkim_result = 'other'
+    res_dkim = {k: 0 for k in ['none', 'neutral', 'pass', 'fail', 'softfail', 'other', 'temperror', 'permerror']}
+    if dkim_result in res_dkim:
+        res_dkim[dkim_result] += 1
+    else:
+        res_dkim['other'] += 1
+    res_dkim['non-positive'] = spf_result in set(['none', 'fail', 'softfail'])
+    list_unsubscribe = get_header_value(email_headers, header_name='List-Unsubscribe')
+    unsubscribe_post = get_header_value(email_headers, header_name='List-Unsubscribe-Post-SPF')
+    res_unsubscribe = 1 if list_unsubscribe is not None or unsubscribe_post is not None else 0
+    from_value = get_header_value(email_headers, header_name='From')
+    return_path_value = get_header_value(email_headers, header_name='Return-Path')
+    last_received_value = get_header_value(email_headers, header_name='Received', index=-1)
+    if from_value is not None:
+        from_address = re.findall(r'<\S+>', from_value)[0][1:-1]
+    else:
+        from_address = ''
+    if return_path_value is not None:
+        return_path_adress = return_path_value.strip().lstrip('<').rstrip('>').strip()
+    else:
+        return_path_adress = ''
+    if last_received_value is not None:
+        received_address_list = re.findall(r'(?<=from )\S+', last_received_value)
+        if len(received_address_list) > 0:
+            received_address = received_address_list[0]
+        else:
+            received_address = ''
+    else:
+        received_address = ''
+    try:
+        from_domain = from_address.split('@')[-1].split('.')[-2] if from_address != '' else ''
+    except Exception:
+        print('error on from_address: {}'.format(from_address))
+        from_domain = ''
+
+    if from_address == '':
+        return_path_same_as_from = float('nan')
+    else:
+        return_path_same_as_from = from_address == return_path_adress
+
```

Comment:
add some more features as mentioned
return path domain as from domain
reply to as from
reply to domain as from
reply to as return path
reply to domain as return path domain
reply to exists